bin/supervise.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+
+logger()
+{
+    level=${1:-info}
+    if [[ -n $JQ ]]; then
+        jq --raw-input --raw-output --compact-output \
+           --arg status "$level" \
+           '{
+                # theres no jq way I can find to format timestamp w/ subsecond precision
+                # so here we are, rolling our own!
+                timestamp: now |
+                           {
+                               # just the date-time stamp w/o Z
+                               datetime: . | strftime("%Y-%m-%dT%H:%M:%S"),
+                               # ms part of fractional seconds w/o leading 0
+                               ms: . | modf | .[0] | "\(.)" | .[1:5]
+                           } |
+                           "\(.datetime)\(.ms)Z",
+                status: $status,
+                message: .
+            }'
+    else
+        # provide a _very_ minimal fallback here
+        read message
+        echo "{\"status\":\"$level\",\"message\":\"$message\"}"
+    fi
+}
```

Comment:
I know we internally use structured JSON logs but we don't have to use structured logs in this script. I think structured logs aren't providing a benefit here.

Reply:
yeah, that's fair.  I wanted to have some way to include debug-level logs and that requires some kind of log level mechanism.  timestamps also provide useful information when monitoring the performance of our termination behavior (we want to know when the termination signal was received vs. when the various shutdown behavior happen).